client/app/scripts/components/node-details/node-details-image-status.js

```diff
@@ -0,0 +1,94 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import find from 'lodash/find';
+import map from 'lodash/map';
+import { CircularProgress } from 'weaveworks-ui-components';
+
+import { getImagesForService } from '../../actions/app-actions';
+
+const topologyWhitelist = ['services'];
+
+function getNewImages(images, currentId) {
+  // Assume that the current image is always in the list of all available images.
+  // Should be a safe assumption...
+  const current = find(images, i => i.ID === currentId);
+  const timestamp = new Date(current.CreatedAt);
+  return find(images, i => timestamp < new Date(i.CreatedAt)) || [];
+}
+
+class NodeDetailsImageStatus extends React.PureComponent {
+  constructor(props, context) {
+    super(props, context);
+
+    this.handleServiceClick = this.handleServiceClick.bind(this);
+  }
+
+  componentDidMount() {
+    if (this.props.serviceName) {
+      this.props.getImagesForService(this.props.params.orgId, this.props.serviceName);
```

Comment:
It's a bit confusing that what we call `serviceName` here is called `serviceId` in `getImagesForService`, so I'd just decide on one name..